pkg/network/cluster_config.go

```diff
@@ -206,18 +211,43 @@ func validateOVNKubernetesCIDROverlap(clusterCofnig *configv1.Network, operConfi
 	return nil
 }
 
+const metricLiveMigrationBlockedLabelKey = "reason"
+
+// openshift_network_operator_live_migration_blocked metric 'reason' label key name values
+const (
+	unsupportedCNI                     = "UnsupportedCNI"
+	unsupportedHCPCluster              = "UnsupportedHyperShiftCluster"
+	unsupportedSDNNetworkIsolationMode = "UnsupportedSDNNetworkIsolationMode"
+	unsupportedMACVlanInterface        = "UnsupportedMACVLANInterface"
+)
+
+var metricLiveMigrationBlocked = metrics.NewGaugeVec(&metrics.GaugeOpts{
+	Namespace: "openshift_network_operator",
+	Name:      "live_migration_blocked",
+	Help: "A metric which contains a constant '1' value labeled with the reason CNI live migration may not begin. " +
+		"The metric is available when CNI live migration has started by annotating the Network CR.",
+}, []string{metricLiveMigrationBlockedLabelKey})
+
+var liveMigrationBlockedMetricOnce sync.Once
+
 func validateLiveMigration(clusterConfig *configv1.Network, infraRes *bootstrap.InfraStatus, client cnoclient.Client) error {
+	liveMigrationBlockedMetricOnce.Do(func() {
+		legacyregistry.MustRegister(metricLiveMigrationBlocked)
+	})
+	metricLiveMigrationBlocked.Reset()
```

Comment:
This removes the metric right?
We need to remove if any of the migration blocking conditions were removed.

Reply:
yes, it removes all labels that were previously set. 
> We need to remove it if any of the migration blocking conditions were removed.

indeed